go/libraries/doltcore/sqle/dtables/commit_diff_table.go

```diff
@@ -151,11 +151,15 @@ func (dt *CommitDiffTable) Partitions(ctx *sql.Context) (sql.PartitionIter, erro
 }
 
 func (dt *CommitDiffTable) LookupPartitions(ctx *sql.Context, i sql.IndexLookup) (sql.PartitionIter, error) {
-	if len(i.Ranges) != 1 || len(i.Ranges[0]) != 2 {
+	ranges, ok := i.Ranges.(sql.MySQLRangeCollection)
+	if !ok {
+		return nil, fmt.Errorf("commit diff table requires MySQL ranges")
```

Comment:
What's the plan here? Is the range abstraction in the GMS PR incomplete and will be added to later, so that we can use it here and other places that now require a type cast? Or do we need to re-implement range semantics everywhere a Doltgres range type is present?